sdk/storage/storage-file-datalake/review/storage-file-datalake.api.md

```diff
@@ -294,6 +302,21 @@ export interface FileFlushOptions extends CommonOptions {
     retainUncommittedData?: boolean;
 }
 
+// @public
+export interface FileParallelUploadOptions extends CommonOptions {
+    abortSignal?: AbortSignalLike;
+    chunkSize?: number;
+    close?: boolean;
+    conditions?: DataLakeRequestConditions;
+    maxConcurrency?: number;
+    metadata?: Metadata;
+    onProgress?: (progress: TransferProgressEvent) => void;
+    pathHttpHeaders?: PathHttpHeaders;
+    permissions?: string;
+    singleUploadThreshold?: number;
```

Comment:
@bterlson Renamed `initialTransferSize` to `singleUploadThreshold`.
When data size <= the limit, will do a single upload.
When data size > limit, all data will be uploaded in chunk of chunkSize in parallel.